src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -1744,10 +1744,31 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     // === Project Name Editing ===
     // ============================
 
+
+    // === Start project name edit ===
+    frp::extend! { network
+        edit_mode     <- bool(&inputs.edit_mode_off,&inputs.edit_mode_on);
+        start_editing <- model.breadcrumbs.project_mouse_down.gate(&edit_mode);
+        eval_ start_editing ( model.breadcrumbs.start_project_name_editing.emit(()) );
+
+        hovered_while_edit_mode <- all(edit_mode,model.breadcrumbs.project_name_hovered)
+            .map(|(edit_mode,hovered)| *edit_mode && *hovered);
+        project_name_edit_hover_start <- hovered_while_edit_mode.gate(&hovered_while_edit_mode);
+        project_name_hover_style      <- project_name_edit_hover_start
+                                            .constant(cursor::Style::new_text_cursor());
+        project_name_edit_hover_end   <- hovered_while_edit_mode.gate_not(&hovered_while_edit_mode);
+        project_name_hover_style_end  <- project_name_edit_hover_end
+                                            .constant(cursor::Style::default());
+        project_name_hover_style      <- any(project_name_hover_style,project_name_hover_style_end);
+
+    }
```

Comment:
I don't like this logic is in the graph editor. Basically, the graph editor knows about `edit_mode` (it's OK), it should send this signal to `breadcrumbs`. But the whole logic like starting editing on project name click (etc.) should be incorporated inside of `breadcrumbs`, as this is the logic of this component, this is not a logic of graph editor. The same applies to mouse styles - `breadcrumbs` should have cursor style output, which could be merged into the list of styles of other components in the graph editor, but this is not the responsibility of the graph editor to contain the logic of when to change cursor on some interaction with bread rumbs panel.

Reply:
Refactored.